server/index.d.ts

```diff
@@ -1,32 +1,25 @@
-import BaseServer, { Logger, Reporter, BaseServerOptions } from '../base-server'
+import BaseServer, { Logger, BaseServerOptions } from '../base-server'
 
-export type ServerOptions = BaseServerOptions & {
+export type LoggerOptions = {
   /**
-   * Custom reporter for process/errors. You should use it only for test purposes
-   * or unavoidable hacks.
-   *
-   * ```js
-   * new Server({
-   *   …,
-   *   reporter: (name, details) => {
-   *     console.log('Event:', name)
-   *     console.log('Details:', JSON.stringify(details))
-   *   }
-   * })
-   * ```
+   * Logger message format.
    */
-  reporter?: Reporter
+  type?: 'human' | 'json'
```

Comment:
Custom logger is excluded from here, cause it conflicts with stream option. (It will not respect it)

Reply:
We will need a way to add, for instance, [Google Cloud pino driver](https://github.com/ovhemert/pino-stackdriver/blob/master/docs/API.md).

Do we have the way without using `server.on('report'`?

Reply:
We can pass a custom logger, as described in docs, since we still have `logger?: Logger | LoggerOptions`
   ```js
    const pino = require('pino')
    const pinoElastic = require('pino-elasticsearch')
   
    const streamToElastic = pinoElastic({
      consistency: 'one',
      node: 'http://localhost:9200',
      ecs: true
    })
   
    const server = new Server(
      Server.loadOptions(process, {
        …,
        logger: pino({ level: 'info' }, streamToElastic)
      })
    )
```

`LoggerOptions` allows us to customize built-in logger.
Besides most of pino custom drivers differs only in stream as in example above, so we can do:

```js
const pinoElastic = require('pino-elasticsearch')
   
const streamToElastic = pinoElastic({
  consistency: 'one',
  node: 'http://localhost:9200',
  ecs: true
})
   
const server = new Server(
  Server.loadOptions(process, {
    …,
    logger: { stream: streamToElastic }
  })
)
```

with google it should be:
```js
const stackdriver = require('pino-stackdriver')

// create the stackdriver destination stream
const credentials = '/credentials.json'
const projectId = 'my-project'
const writeStream = stackdriver.createWriteStream({ credentials, projectId })

const server = new Server(
  Server.loadOptions(process, {
    …,
    logger: { stream: writeStream }
  })
)
```